src/gwt/src/org/rstudio/studio/client/projects/ProjectOpener.java

```diff
@@ -49,14 +62,37 @@ public void showOpenProjectDialog(
             fsContext, 
             FileSystemItem.createDir(defaultLocation),
             "R Projects (*.Rproj)",
+            true,
             new ProgressOperationWithInput<FileSystemItem>()
             {
                @Override
                public void execute(FileSystemItem input,
-                     ProgressIndicator indicator)
+                                   final ProgressIndicator indicator)
                {
-                  onCompleted.execute(new OpenProjectParams(input, null, false), 
-                        indicator);
+                  if (input.isDirectory())
+                  {
+                     String rprojPath = input.completePath(input.getName() + ".Rproj");
+                     input = FileSystemItem.createFile(rprojPath);
+                  }
+                  
+                  final FileSystemItem projFile = input;
+                  server_.createProjectFile(
```

Comment:
This should be a no-op in that case, but I agree we might as well avoid making an excessive RPC request here.